src/main/java/rx/Single.java

```diff
@@ -1789,4 +1790,36 @@ public void onNext(T t) {
         return zip(this, other, zipFunction);
     }
 
+    /**
+     * Modifies the source {@link Single} so that it invokes an action when it calls {@code onSuccess}.
+     * <p>
+     * <img width="640" height="310" src="https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/doOnNext.png" alt="">
+     * <dl>
+     *  <dt><b>Scheduler:</b></dt>
+     *  <dd>{@code doOnSuccess} does not operate by default on a particular {@link Scheduler}.</dd>
+     * </dl>
+     *
+     * @param onSuccess
+     *            the action to invoke when the source {@link Single} calls {@code onSuccess}
+     * @return the source {@link Single} with the side-effecting behavior applied
+     * @see <a href="http://reactivex.io/documentation/operators/do.html">ReactiveX operators documentation: Do</a>
+     */
+    public final Single<T> doOnSuccess(final Action1<? super T> onSuccess) {
```

Comment:
I didn't add `@Experimental` here and in the other PRs because whole class  `Single` is marked with `@Experimental`. Do you think we should mark these basic operators too?